exporters/committime/collector_bitbucket.py

```diff
@@ -1,11 +1,160 @@
+import requests
+import json
+import logging
+import pelorus
 from collector_base import AbstractCommitCollector
+# import urllib3
+# urllib3.disable_warnings()
 
 
 class BitbucketCommitCollector(AbstractCommitCollector):
 
+    # globals for Bitbucket v1 API
+    V1_API_ROOT = 'rest/api'
+    V1_API_TEST = '1.0/projects'
+    V1_API_PATTERN = '1.0/projects/{group}/repos/{project}/commits/{commit}'
+    # globals for Bitbucket v2 API
+    V2_API_ROOT = 'api'
+    V2_API_TEST = '2.0/repositories'
+    V2_API_PATTERN = '2.0/repositories/{group}/{project}/commit/{commit}'
+    # Default http headers needed for API calls
+    DEFAULT_HEADERS = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+
     def __init__(self, kube_client, username, token, namespaces, apps):
-        super().__init__(kube_client, username, token, namespaces, apps, "BitBucket", "TODO-DateFormat")
+        super().__init__(kube_client, username, token, namespaces, apps, 'BitBucket', '%Y-%m-%dT%H:%M:%S%z')
+        self.__server_dict = {}
+        self.__session = requests.Session()
 
+    # base class impl
     def get_commit_time(self, metric):
         """Method called to collect data and send to Prometheus"""
-        print("BitBucket is not yet supported")
+
+        git_server = metric.git_server
+        # do a simple check for hosted Git services.
+        if "github" in git_server or "gitlab" in git_server:
```

Comment:
Is there some more reliable way to determine that our URL is not a bitbucket URL?

Reply:
Do we handle this as an enhancement?  I only suggest that as this should be refactored in the base and will require changes in all the commit exporters.